rdr_service/resource/generators/_base.py

```diff
@@ -61,7 +63,11 @@ def _get_or_create_type_record(self, dao, schema_meta):
             rec.typeUID = type_uid
 
             session.add(rec)
-            session.commit()
+            try:
+                session.commit()
+            except IntegrityError:
+                # Record already exists, parallel tasks can cause this.  We should not see these often.
+                logging.warning(f'Resource type record already exists for {rec.typeName}.')
```

Comment:
What happens to the resource when the type already exists? Should this attempt to return it?

Reply:
This is to handle an edge case where the Type record does not exists but two parallel processes are trying to create it at the same time, which happened during testing on Sandbox.  This does need fixing to try retrieving the record in the Try block.